gateways/orbital/orbital.go

```diff
@@ -0,0 +1,159 @@
+	return &OrbitalClient{
+		host:        orbitalHost(env),
+		credentials: credentials,
+		httpClient:  httpClient,
+	}
+}
+
+func (client *OrbitalClient) Authorize(request *sleet.AuthorizationRequest) (*sleet.AuthorizationResponse, error) {
+	authRequest := buildAuthRequest(request, client.credentials)
+
+	orbitalResponse, err := client.sendRequest(authRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	if orbitalResponse.Body.ProcStatus != ProcStatusSuccess {
+		if orbitalResponse.Body.RespCode != "" {
+			errorCode := orbitalResponse.Body.RespCode
+			return &sleet.AuthorizationResponse{
+				Success:   false,
+				ErrorCode: errorCode,
+			}, nil
+		}
+
+		return &sleet.AuthorizationResponse{
+			Success:   false,
+			ErrorCode: RespCodeNotPresent,
+		}, nil
+	}
+
+	if orbitalResponse.Body.RespCode != RespCodeApproved {
+		errorCode := orbitalResponse.Body.RespCode
+		return &sleet.AuthorizationResponse{
+			Success:   false,
+			ErrorCode: errorCode,
+		}, nil
+	}
+
+	return &sleet.AuthorizationResponse{
+		Success:              true,
+		TransactionReference: strconv.Itoa(orbitalResponse.Body.TxRefNum),
+		AvsResult:            translateAvs(orbitalResponse.Body.AVSRespCode),
+		CvvResult:            translateCvv(orbitalResponse.Body.CVV2RespCode),
+		Response:             strconv.Itoa(int(orbitalResponse.Body.ApprovalStatus)),
+		AvsResultRaw:         string(orbitalResponse.Body.AVSRespCode),
+		CvvResultRaw:         string(orbitalResponse.Body.CVV2RespCode),
+	}, nil
+}
+
+func (client *OrbitalClient) Capture(request *sleet.CaptureRequest) (*sleet.CaptureResponse, error) {
+	captureRequest := buildCaptureRequest(request, client.credentials)
+
+	orbitalResponse, err := client.sendRequest(captureRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	if orbitalResponse.Body.ProcStatus != ProcStatusSuccess {
+		if orbitalResponse.Body.RespCode != "" {
+			errorCode := orbitalResponse.Body.RespCode
+			return &sleet.CaptureResponse{
+				Success:   false,
+				ErrorCode: &errorCode,
+			}, nil
+		}
+
+		errorCode := RespCodeNotPresent
+		return &sleet.CaptureResponse{
+			Success:   false,
+			ErrorCode: &errorCode,
+		}, nil
+	}
+
+	if orbitalResponse.Body.RespCode != RespCodeApproved {
+		errorCode := orbitalResponse.Body.RespCode
+		return &sleet.CaptureResponse{
+			Success:   false,
+			ErrorCode: &errorCode,
+		}, nil
+	}
+
+	return &sleet.CaptureResponse{
+		Success:              true,
+		TransactionReference: strconv.Itoa(orbitalResponse.Body.TxRefNum),
+	}, nil
+}
+
+func (client *OrbitalClient) sendRequest(data Request) (*Response, error) {
+
+	bodyXML, err := xml.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+
+	bodyWithHeader := xml.Header + string(bodyXML)
+	reader := bytes.NewReader([]byte(bodyWithHeader))
+	request, err := http.NewRequest(http.MethodPost, client.host, reader)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Add("MIME-Version", "1.1")
+	request.Header.Add("Content-Type", "application/PTI80")
```

Comment:
maybe we make this a const so its more clear where this value comes from? And MIME Version?

Reply:
Made all the headers consts